pkg/provider/provider.go

```diff
@@ -50,6 +50,18 @@ const (
 	skipMultusConnectivityTestsLabel = "test-network-function.com/skip_multus_connectivity_tests"
 )
 
+type PodStruct v1.Pod
```

Comment:
can you try using this 
```
type PodStruct v1.Pod
```
it will simplify the casting you're doing in suite.go

Reply:
yes, this is needed, otherwise we get the object copy where we need the pointer to the object. Right now the skip functionality no longer works

Reply:
just tried, seems like we cannot have a pointer or interface type as a method receiver :-(
I think we should just skip these maps

Reply:
Just tried it, simply using the pointer lead to an error in the following function:
func (p PodStruct) MarshalText() ([]byte, error) {
this is because the receiver object p type should not be a pointer type :-(
I think that we should just skip serializing those 3 maps:
MultusIPs          map[*PodStruct]map[string][]string
	SkipNetTests       map[*PodStruct]bool
	SkipMultusNetTests map[*PodStruct]bool

Reply:
Got it working! we can use:
type PodStruct struct{
BasePod	*v1.Pod
}